pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -231,21 +231,33 @@ impl CustomScan for PdbScan {
                 builder.custom_private().set_quals(quals);
                 builder.custom_private().set_limit(limit);
 
-                if is_topn {
+                if is_topn && pathkey.is_some() {
+                    let pathkey = pathkey.as_ref().unwrap();
                     // sorting by a field only works if we're not doing const projections
                     // the reason for this is that tantivy can't do both scoring and ordering by
                     // a fast field at the same time.
                     //
                     // and sorting by score always works
-                    match (maybe_needs_const_projections, &pathkey) {
-                        (false, Some(pathkey @ OrderByStyle::Field(..))) => {
+                    match (maybe_needs_const_projections, pathkey) {
+                        (false, OrderByStyle::Field(..)) => {
                             builder.custom_private().set_sort_info(pathkey);
+
+                            // Always set the sort info and mark as partial sort if we can handle any pathkey
+                            // whether it's a single pathkey or one of multiple pathkeys
+                            builder.custom_private().set_is_partial_sort(true);
                         }
-                        (_, Some(pathkey @ OrderByStyle::Score(..))) => {
+                        (_, OrderByStyle::Score(..)) => {
                             builder.custom_private().set_sort_info(pathkey);
+
+                            // Always set the sort info and mark as partial sort if we can handle any pathkey
+                            // whether it's a single pathkey or one of multiple pathkeys
+                            builder.custom_private().set_is_partial_sort(true);
                         }
                         _ => {}
                     }
+
+                    // Even for TopN cases, add the pathkey to ensure Incremental Sort gets considered
+                    builder = builder.add_path_key(pathkey);
```

Comment:
This isn't safe to do if parallel workers might be used: see the comments between 366 and 371.

But also: I think that this is already being done (if it is safe) on lines 372 to 379?

Reply:
You are right. Fixed in here: https://github.com/paradedb/paradedb/pull/2455/commits/a5ae808995f2d56667c0aebedccb5354838520a7